### ISSUE TITLE
ACF non-text type field data not showing up in dynamic widgets like Icon URL or Image Src.

### ISSUE THREAD

**Issue by levi-beers** [pojome/elementor#5344]
## Prerequisites

- [ x ] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [ x ] The issue still exists against the latest stable version of Elementor.

## Description

I've been successful in getting my ACF fields with text type to show up in Text/Heading Elementor Widgets using "Dynamic ACF Field", however no other field types are working. Specifically URL & Image. If you choose a button or an icon widget in Elementor and select "Dynamic" for link, then "ACF Field" and then your ACF URL Field type, the link doesn't work when viewing the site. If you choose an image widget in Elementor and select "Dynamic" for image, then "ACF Field" and then your ACF Image Field type, the image doesn't show. 

## Steps to reproduce

1. Install Elementor & Elementor PRO
2. Install ACF Pro 5.7.1
3. Add "Options" page to functions.php
4. Create "Field Group" with text field, image field and URL field. Enter some dumy data.
5. Create an Elementor Page using a heading widget, image widget and button widget -- selecting ACF Field Data as the "Dynamic" type. 

## Isolating the problem

- [ ] This bug happens with only Elementor plugin active (and Elementor Pro). *
- [ x ] This bug happens with a default WordPress theme active.
- [ x ] I can reproduce this bug consistently using the steps above.

* ACF Pro has to be enabled for ACF to work. 

## Environment

<details>
<summary>System Info</summary>
```
					
== Server Environment ==
	Operating System: Linux
	Software: nginx/1.10.1
	MySQL version: 5.6.37
	PHP Version: 7.1.7
	PHP Max Input Vars: 4000
	PHP Max Post Size: 1000M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 4.9.8
	Site URL: http://universal.dev
	Home URL: http://universal.dev
	WP Multisite: No
	Max Upload Size: 300 MB
	Memory limit: 40M
	Permalink Structure: /%postname%/
	Language: en-US
	Timezone: 0
	Debug Mode: Inactive

== Theme ==
	Name: EZTOUSE
	Version: 2.5
	Author: Levi Beers
	Child Theme: Yes
	Parent Theme Name: OceanWP
	Parent Theme Version: 1.5.22
	Parent Theme Author: Nick

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_5) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/11.1.1 Safari/605.1.15

== Active Plugins ==
	Advanced Custom Fields: ACF Code Field
		Version: 1.7
		Author: Peter Tasker

	Advanced Custom Fields PRO
		Version: 5.7.1
		Author: Elliot Condon

	All-in-One WP Migration
		Version: 6.73
		Author: ServMask

	CSS & JavaScript Toolbox
		Version: 8.4.1
		Author: Wipeout Media

	Disable Comments
		Version: 1.7.1
		Author: Samir Shah

	Duplicate Page
		Version: 2.7
		Author: mndpsingh287

	Elementor
		Version: 2.1.6
		Author: Elementor.com

	Elementor Pro
		Version: 2.1.0
		Author: Elementor.com

	EWWW Image Optimizer
		Version: 4.2.3
		Author: Shane Bishop

	EZ Elementor Additions
		Version: 1.0.0
		Author: EZToUse.com

	EZ Preview Auto-Login
		Version: 1.0
		Author: EZToUse.com

	Formidable Forms
		Version: 3.03
		Author: Strategy11

	Formidable Forms Pro
		Version: 3.03
		Author: Strategy11

	Gutenberg
		Version: 3.4.0
		Author: Gutenberg Team

	Mailgun
		Version: 1.5.11
		Author: Mailgun

	Ocean Elementor Widgets
		Version: 1.1.0-beta.3
		Author: OceanWP

	Ocean Extra
		Version: 1.4.20
		Author: OceanWP

	Ocean Hooks
		Version: 1.1.0-beta.1
		Author: OceanWP

	Ocean White Label
		Version: 1.0.5
		Author: OceanWP

	Stop Spammers
		Version: 7.1
		Author: Bryan Hadaway

	WooCommerce
		Version: 3.4.4
		Author: Automattic

	Yoast SEO
		Version: 7.9
		Author: Team Yoast


== Debug ==
	Errors: There are no errors to display
				
```
</details>

**Comment by levi-beers** [comment]
@bainternet this issue isn't resolved in Elementor 2.1.7 and Elementor PRO 2.1.3. If you create a URL field on your options page and enter a website, and then drop an icon widget on a page, select "dynamic link" and choose "ACF URL Field (Options:url_field_name) and view the page, the link doesn't show up. Same with an image. I can make a video in the morning that shows the issue for clarity, but it still definitely exists.

**Comment by levi-beers** [comment]
I updated to Elementor 2.1.8 and Elementor PRO 2.1.4 and this issue still persists.

**Comment by bainternet** [comment]
@levi-beers 

>I can make a video in the morning that shows the issue for clarity, but it still definitely exists.
can you please that would help.

**Comment by phoenix23590** [comment]
Resolve!
In acf-image.php and acf-url.php, change:
`$field = get_field_object( $field_key );`
to:
`if ( 'options' === $field_key ) {
$field = get_field_object( $meta_key, $field_key );
		} else {
			$field = get_field_object( $field_key );
		}`

In acf-image.php, change `$field['save_format']` to `$field['return_format']`

Hope this issue are fixed in next update.

**Comment by levi-beers** [comment]
@phoenix23590 I've confirmed on my install this works. The only thing is that if someone has the IMAGE field in ACF set to "array" or "url" it won't work, it only works under "id", which is fine by me but I doubt that's the expected behavior because there are case switches for array & url.

**Comment by levi-beers** [comment]
[ACF Image issue] when custom field is "array" or "url". User @phoenix23590 
 code above fixes issue with ACF URL field and ACF image field type "ID" (https://www.youtube.com/watch?v=S8x-SHqu9x0&feature=youtu.be)

**Comment by phoenix23590** [comment]
@levi-beers "url" works fine for me. In case "array", change `case 'object':` to `case 'array':`

My full code :
> 
            if ( 'options' === $field_key ) {
                $field = get_field_object( $meta_key, $field_key );
            } else {
                $field = get_field_object( $field_key );
            }

			if ( $field && ! empty( $field['return_format'] ) ) {
				switch ( $field['return_format'] ) {
					case 'array':
						$value = $field['value'];
						break;
					case 'url':
						$value = [
							'id' => 0,
							'url' => $field['value'],
						];
						break;
					case 'id':
						$src = wp_get_attachment_image_src( $field['value'], $field['preview_size'] );
						$value = [
							'id' => $field['value'],
							'url' => $src[0],
						];
						break;
				}
			}

**Comment by bainternet** [comment]
@phoenix23590 @levi-beers 

Thanks, should be fixed in next version.

**Comment by Bey13** [comment]
Not 100% sure this is related and this is the only thread I have seen regarding ACF options not showing up. I used to have the option for "ACF Beta" (or something similar) as a Dynamic Content option. However, after updating to the newest stable release, I am not longer seeing that option in any of the Dynamic Content menus in various widgets/elements. 

I only got to play around with the Section Background Image with ACF, however now not seeing that option either. Has this option been pulled for the time being as it was stated to be in "beta"?

Best Regards
C

**Comment by krisblack** [comment]
I'm experiencing the same problem as @Bey13. I had Text widgets placed in a template that would pull data from an ACF form on my projects post type. It was working until a couple versions ago. I have the latest updates for Elementor Pro and my other plugins.

**Comment by Bey13** [comment]
Hey there @krisblack 

Seems one of the newer version of Elementor changed to support only ACF version 5+. I was recommended to upgrade to the early access version of ACF (outlined here - https://www.advancedcustomfields.com/resources/upgrade-guide-version-5/). 

I was having issues triggering the update for ACF version 5 and ended up purchasing the Pro version which is above version 5. I am not able to see the ACF Field option available in the Dynamic Content menus, this time without the "beta" tag with it. 

Hope this helps!

Best Regards,
C

**Comment by krisblack** [comment]
Thanks @Bey13. Upgrading to the ACF 5 beta fixed the issue for me. Your
help is very much appreciated.

Kris


*——*

*Kris Black* ... *designer, author, illustrator*
studio@krisblack.com … @krisblack <http://twitter.com/krisblack>

*Kris Black Studio* LLC
www.krisblack.com

On August 28, 2018 at 4:31:03 PM, Bey13 (notifications@github.com) wrote:

Hey there @krisblack <https://github.com/krisblack>

Seems one of the newer version of Elementor changed to support only ACF
version 5+. I was recommended to upgrade to the early access version of ACF
(outlined here -
https://www.advancedcustomfields.com/resources/upgrade-guide-version-5/).

I was having issues triggering the update for ACF version 5 and ended up
purchasing the Pro version which is above version 5. I am not able to see
the ACF Field option available in the Dynamic Content menus, this time
without the "beta" tag with it.

Hope this helps!

Best Regards,
C

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/pojome/elementor/issues/5344#issuecomment-416730008>,
or mute the thread

**Comment by Bey13** [comment]
Happy that was of use to you @krisblack and got that working for yah!

Best Regards,
C

**Comment by levi-beers** [comment]
In the latest version of Elementor Pro (2.1.7), I still have to manually change modules/dynamic-tags/acf/tags/acf-image.php as mentioned by @phoenix23590 for ACF images to work. I saw this issue number referenced in the changelog for 2.1.6 so I wanted to clarify the image issue still exists in the official Elementor release (2.1.7).

**Comment by levi-beers** [comment]
@Bey13 I should be a little more clear, with Elementor 2.2.1 and Elementor Pro 2.1.7, I still have to manually edit modules/dynamic-tags/acf/tags/acf-image.php as mentioned by @phoenix23590 for ACF images to work. I just grabbed a fresh copy and can confirm, the issue with the image field still exists in the official release.

**Comment by phoenix23590** [comment]
I confirm this issue still occurred. ACF Pro replaced "save_format" to "return_format" in ACF image.
And there is issue in ACF gallery, also. It's should be:
`if ( 'options' === $field_key ) {
                $field = get_field_object( $meta_key, $field_key );
            } else {
                $field = get_field_object( $field_key );
            }`
when get_field_object.

**Comment by bainternet** [comment]
@phoenix23590 

Thanks for reporting gallery fix implemented in upcoming v2.1.9

@levi-beers 

Just checked again and in 2.1.8 there is no need to manually patch.

thanks.

**Comment by Anaon** [comment]
Is it fixed for anyone?

Nothing changed on my side

**Comment by sermalefico** [comment]
Nothing has changed. And I also do not see that the dynamic images work in the templates of page, post and custom taxonomies.

**Comment by bainternet** [comment]
fixed, thanks.

**Comment by Anaon** [comment]
It is not fixed, do we need to open another issue for the same bug?

**Comment by GrandGrand** [comment]
Hi there. Agree with Anaon. Not fixed. Will expand on this in another comments.

**Comment by barrychapman** [comment]
This is not fixed. My fields ONLY return the integer ID. No matter what I have selected, array - url - id.

Can we get this fixed?

**Comment by Anaon** [comment]
Any Update from the team?

**Comment by Anaon** [comment]
Tried again today, still not fixed

**Comment by GrandGrand** [comment]
Recap: There is still no update, and this is raging for this subject should not have been closed on Sptember 20.
Solution: Manual fix 
Author: @phoenix23590 in his Aug 22 post.
How: Manually edit some "Elementor Pro plugin" folder files.
Files location/path: plugins > Elementor Pro > modules > dynamic-tags > acf > tags > acf-image.php

**Comment by phoenix23590** [comment]
@bainternet ACF changed `save_format` to `return_format`. Please update it in acf-image.php

**Comment by zeinnicholas** [comment]
Still no fix

**Comment by bainternet** [comment]
@phoenix23590 

Thanks, already merged in 2.1.12 

Anyone else having an issue Open a new one.

**Comment by sermalefico** [comment]
I think it's not fixed yet. It worked for me only once. After and added another widget stopped working.

**Comment by ericadyson** [comment]
It's still not working for me. Can't get URK or link to work. They do not show in Elementor at all. Help!

**Comment by bainternet** [comment]
Issue in OP is long closed and already merged in 2.1.12

Anyone else having an issue Open a new one.

**Comment by Anaon** [comment]
That's what I asked on 20th of september. Do we really need to open a new issue when it is not fixed ?

There is the history of the home-made fix and the people describing their issue.

Tried again today and it is still NOT fixed

**Comment by sermalefico** [comment]
wtf?!

**Comment by sermalefico** [comment]
Still no fix

**Comment by Anaon** [comment]
it appears that @bainternet doesn't want this to be fixed :/

**Comment by ericadyson** [comment]
Why on earth not?




*Erica Dyson אריקה דייסון *
M: +972-50-544-8044 F: 077-3179527 A: Kibbutz Hamadia
E ericadyson@gmail.com

W: www.ericadyson.com


[image: Mailtrack]
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality5&>
Sender
notified by
Mailtrack
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality5&>
11/28/18,
7:44:39 PM

**Comment by ericadyson** [comment]
I wonder why my comment regarding the url field not showing in Elementor has been deleted. This is still an issue. Please explain why it's been deleted and if fix is due. Thanks

**Comment by depasco** [comment]
This issue is just fixed since the latest update of the pro and the normal version. Before the issue was still there. Just update and the bug is gone.

**Comment by papplebee** [comment]
This issue is not fixed for me. I have disabled all plugins except Elementor, Elementor Pro, and ACF, and I am still experiencing this issue.

**Comment by henrique-voni** [comment]
Still not fixed for me. See: https://prntscr.com/mx2r2t

**Comment by ericadyson** [comment]
And I still have the issue with url .

**Comment by sermalefico** [comment]
Not fixed

**Comment by blogbytes** [comment]
This is still not working and the fix above does nothing on my side. 
Any update on this?

**Comment by sermalefico** [comment]
#5147

**Comment by jgajera** [comment]
+1

My buttons aren't rendering href for the ACF URL field (on author pages, so the link is dynamically generated)

**Comment by jhixson** [comment]
What fixed this for me was changing 
```
$field = get_field_object( $field_key, get_queried_object() );
```
to just:
```
$field = get_field_object( $field_key );
```
To clarify I was using this in a Posts widget loop and I think the function was pulling the wrong post data.

**Comment by Skylinks854** [comment]
I Have Never Faced Any Issue Like This Until Latest Update .. Acf Url Field Is Not Working Anymore.
I Have Tried What @jhixson Suggest But Elementor Is Replicating Same Field Twice .. I,m Using 2 url Fields In A Single Post.

**Comment by jhixson** [comment]
As an update, I was using a plugin (Ele Custom Skin) to pull in custom skins for posts widgets. Once I updated that I was able to get the correct fields again. So I'm not sure the `get_field_object` call was the culprit or not.

**Comment by Infestix** [comment]
I have the exact same problem.  Ele Custom Skin loop and url fields are off. Any ideas how to fix that?

> As an update, I was using a plugin (Ele Custom Skin) to pull in custom skins for posts widgets. Once I updated that I was able to get the correct fields again. So I'm not sure the `get_field_object` call was the culprit or not.

**Comment by emme10** [comment]
It seems that this is still an issue. I cannot display an image definied in ACF as background of a column.
And its also not working when I try to display it as normal elementor image.

Tried all of the above suggestion like changing php code inside the plugin as well as using it as url,array or id but the image is never showing up.

Any idea on a fix? I urgently need this feature for a client.

**Comment by timtunnicliff** [comment]
I am also having this issue. URL and Image ACF's not showing within an Archive.

**Comment by Skylinks854** [comment]
Yes ACF Url not working in Elementor Archive pages, also now i can see code is changed a bit developers have removed field key and old solution posted in this thread not working anymore .. if someone have any solution to it then please share with us thank you!

**Comment by vitaminn** [comment]
having same issue here.

**Comment by Mulli** [comment]
Cannot display video (url) nor embed video in text editor widget using ACF PRO.
WP 5.6, Elementor 3.0.16  Pro 3.0.9, ACF PRO 5.9.3 (all latest as of Jan 10, 2021)

**Comment by latheeshvmv** [comment]
The issue is back again

**Comment by GalHadas12** [comment]
Elementor don't recognize ACF image field

**Comment by svrdoljak123** [comment]
What's the deal... Elementor I have sent two e-mails to support... still no solution. ACF fields and even Crockoblock custom type fields do not work anymore if you use dynamic tags and select a field to display. WTF?

**Comment by EvilBox** [comment]
The problem is several years old! And there is still no solution, why?

**Comment by BinTofajjal** [comment]
Still having the same issue text fields works but Taxonomy is not working...

**Comment by selejanalex** [comment]
The problem is still there. The iframe (Youtube video) embeded in the Wordpress Editor Classic does not render when using Dynamic tags.

**Comment by robtucker168** [comment]
Image problem is still there. I can only call ACF User 'Text' Fields using the the dynamic 'User Meta' option. Can't call ACF images, or number fields (for counter widget)

**Comment by BlazPerus** [comment]
Cmon Elementor, this are basics. I'm literally 1cm away from not using Elementor again.

**Comment by manuell18** [comment]
As of 2025, Elementor has yet to resolve this issue.
